mono/mini/method-to-ir.c

```diff
@@ -3981,6 +3981,8 @@ handle_delegate_ctor (MonoCompile *cfg, MonoClass *klass, MonoInst *target, Mono
 	/* Set target field */
 	/* Optimize away setting of NULL target */
 	if (!MONO_INS_IS_PCONST_NULL (target)) {
+		MONO_EMIT_NEW_BIALU_IMM (cfg, OP_COMPARE_IMM, -1, target->dreg, 0);
```

Comment:
I thought so too and tried it. Turns out it doesn't do anything unless explicit null checks are enabled:
https://github.com/mono/mono/blob/492438a431ea5bf563735055441cbe81a453b195/mono/mini/ir-emit.h#L889